core/src/main/scala/akka/persistence/cassandra/journal/CassandraJournal.scala

```diff
@@ -414,59 +414,71 @@ class CassandraJournal(cfg: Config) extends AsyncWriteJournal
+        deleteResult.failed.foreach { e =>
           log.warning("Unable to complete deletes for persistence id {}, toSequenceNr {}. " +
             "The plugin will continue to function correctly but you will need to manually delete the old messages. " +
             "Caused by: [{}: {}]", persistenceId, toSequenceNr, e.getClass.getName, e.getMessage)
         }
+        deleteResult.map(_ => Done)
       }
     }
 
     /**
-     * Deletes the by inserting into the metadata table deleted_to
+     * Deletes the events by inserting into the metadata table deleted_to
      * and physically deletes the rows.
-     *
-     * The Future completes once the logical delete is complete as
-     * that is when future reads won't see the events.
      */
-    def logicalDeleteWithAsyncPhysicalDelete(lowestSequenceNr: Long, highestSequenceNr: Long): Future[Done] = {
-      val lowestPartition = partitionNr(lowestSequenceNr, config.targetPartitionSize)
+    def logicalDeleteWithAsyncPhysicalDelete(highestDeletedSequenceNumber: Long, highestSequenceNr: Long): Future[Done] = {
+      val lowestPartition = partitionNr(highestDeletedSequenceNumber + 1, config.targetPartitionSize)
       val toSeqNr = math.min(toSequenceNr, highestSequenceNr)
       val highestPartition = partitionNr(toSeqNr, config.targetPartitionSize) + 1 // may have been moved to the next partition
-      val boundInsertDeletedTo = preparedInsertDeletedTo.map(_.bind(persistenceId, toSeqNr: JLong))
-      val logicalDelete = boundInsertDeletedTo.flatMap(session.executeWrite)
-      // Done async as thw plugin won't see the events once the logical delete is complete
-      logicalDelete.foreach(_ => physicalDelete(lowestPartition, highestPartition, toSeqNr))
-      logicalDelete.map(_ => Done)
+      val logicalDelete =
+        if (toSeqNr <= highestDeletedSequenceNumber) {
+          // already deleted same or higher sequence number, don't update highestDeletedSequenceNumber,
+          // but perform the physical delete (again), may be a retry request
+          Future.successful(())
+        } else {
+          val boundInsertDeletedTo = preparedInsertDeletedTo.map(_.bind(persistenceId, toSeqNr: JLong))
+          boundInsertDeletedTo.flatMap(session.executeWrite)
+        }
+      logicalDelete.flatMap(_ => physicalDelete(lowestPartition, highestPartition, toSeqNr))
     }
 
     val logicalDelete = for {
       highestDeletedSequenceNumber <- asyncHighestDeletedSequenceNumber(persistenceId)
-      lowestSequenceNr <- asyncReadLowestSequenceNr(persistenceId, 1L, highestDeletedSequenceNumber, Some(config.readConsistency), Some(readRetryPolicy))
-      highestSequenceNr <- asyncFindHighestSequenceNr(persistenceId, lowestSequenceNr, config.targetPartitionSize)
-      _ <- logicalDeleteWithAsyncPhysicalDelete(lowestSequenceNr, highestSequenceNr)
+      highestSequenceNr <- {
+        // MaxValue may be used as magic value to delete all events without specifying actual toSequenceNr
```

Comment:
I have also added a check of valid seqnr in Akka https://github.com/akka/akka/pull/25488